### PR TITLE
Cu 86bxxf09 page on boarding

### PR DESCRIPTION
**Pull request by MarvinFlm**
# Description

Add an onBoarding to enter the information of the patient

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [X] I have assigned this PR to myself
- [X] I have added at least 1 reviewer
- [X] I have added the needed labels
- [X] I have linked this PR to a clickup task
- [X] I have tested this code
- [ ] I have added / updated tests (unit / functionals / end-to-end / ...)
- [ ] I have updated the README and other relevant documents (guides...)
- [ ] I have added sufficient documentation both in code, as well as in the READMEs

### CODE REVIEW

**Review by lucas-louis — CHANGES_REQUESTED** [review]
Banner:
- Maybe add more padding on the top of the banner => not enough space on screen without native top bar
- The Edgar is not the good one

- Not enough spacing between the steps and the fields
- You need to update the placeholder of inputs
- The button for the sex and the health issues are not well designed
- Missing uppercase on "Oui" and "Non" button
- The bottom button is too small

- There is no error message (like when you try to continue without filling all the inputs)
- The birthdate needs to be locked after the day of today => you can't be born in the future from now
- When you put a birthdate and you change the sex or the health issues, the birthdate is cleared
- The height and the weight needs to be number input and not text input

⚠️ I can't go to the other steps

⚠️ Is the signup page really working!!

**Review by lucas-louis — CHANGES_REQUESTED** [review]
- I don't think that the signup/ login page are working
- The button for the sex and the health issues are not well designed
- Missing uppercase on "Oui" and "Non" button
- There is no error message (like when you try to continue without filling all the inputs)
- When you put a birthdate and you change the sex or the health issues, the birthdate is cleared
- Once the "previous" button is clicked we can't continue the onboarding
- The doctor step doesn't show anything

**Review by lucas-louis — CHANGES_REQUESTED** [review]
- The value of the input needs to start with an uppercase by default
- The button for the sex and the health issues are not well designed (no white background?)
- The "Oui" and "Non" buttons do not follow the design
- Missing padding at the bottom (the continue button is too near to the last input)
- The doctor step doesn't show anything (AGAIN)

**Review by lucas-louis — CHANGES_REQUESTED** [review]
- The value of the input needs to start with an uppercase by default (AGAIN)
- The button for the sex and the health issues are not well designed (no white background?) (AGAIN)
- The "Oui" and "Non" buttons do not follow the design (AGAIN)
- Missing padding at the bottom (the continue button is too near to the last input) (AGAIN)

Doctor selection:
- No error message when clicking on the next button when no doctor is selected
- Is it connected to the backend?

Antecedents:
- Grey square under the button??
- The selection of the antecedents doesn't work
- The modal needs to open when the input is clicked

**Review by lucas-louis — CHANGES_REQUESTED** [review]
## The register doesn't create an account

Doctor selection:
- No error message when clicking on the next button when no doctor is selected
- The search bar need to filter without "validating" the search (dynamic search)
- Bad icon for the search bar
- Missing label over the search bar
- Too much spacing between the search bar and the doctor cards
- Too big arrow for the doctor cards
- The name of the doctor needs to follow the design
- The address needs to follow the design

Antecedents:
- You can't validate without adding antecedents (if you say that you have antecedents)
- Bad icon on antecedent input
- "Ajoutez" instead of "Ajouter"
- The font weights are not good
- Is there a shadow under the button of the still relevant value?
- "Nom de votre" instead of "Nom du votre"
- Too much padding at the bottom of the drawer
- You can't delete an added antecedent

Treatment
- Bad icon on drawer
- Bad title on drawer
- Bad icon on input
- The font weights are not good
- "Jour de prise" instead of "Jour des prises"
- "Période de prise" instead of "
- Maybe make the day/ period a bit higher
- The medicines selector is really bad:
  - Too big on some screens
  - If I close the keyboard the selector closes
  - I can't scroll in the selector
- You can't add a medicine without at least a period, a day, a quantity and a name
- If you add another medicine, it'll always be the same medicine name at the end
- You can't remove an added medicine
- You can't add an antecedents without information
- The scroll is weird when there is more than one medicine

Displaying the info of added antecedents
- The font weights are not good
- The medicine icon needs to be blue if there is medicines linked to the antecedent
- You can't display the information of the antecedents if there is no medicines linked to it
- Too much spacing between the label and the list of medicines
- You need to add an "s" if there is more than 1 unit of medicine
- The border of the title needs to be 1px width
- Too much spacing between the days and periods
- Is there a shadow under the button of the still relevant value?

**Review by lucas-louis — CHANGES_REQUESTED** [review]
Doctor selection:
- Too much spacing between the search bar and the doctor cards (good for now)
- Bad font for the address of the doctors

Antecedents:
- Bad font weight for the labels (too thin)
- Title of the modal too bold?
- Placeholder of the add medicine => "Ajouter" instead of "Ajoutez"
- Too much padding at the bottom of the drawer
You can't delete an added antecedent

Treatment
-  "Ajouter" instead of "Ajoutez"
- "Nom de votre" instead of "Nom du votre"
- The medicines selector is really bad:
  - Too big on some screens
  - If I close the keyboard the selector closes
  - I can't scroll in the selector
- If you add another medicine, it'll always be the same medicine name at the end
- The scroll is weird when there is more than one medicine

Displaying the info of added antecedents
- You can't display the information of the antecedents if there is no medicines linked to it
- You need to add an "s" if there is more than 1 unit of medicine
- The border of the title needs to be 1px width

**Review by lucas-louis — CHANGES_REQUESTED** [review]
Signup page:
- When bad email address, redirected to the onboarding

Informations page:
- Yes/ No button => font weight too bold

Doctor selection:
- Too much spacing between the search bar and the doctor cards (good for now)

Antecedents:
- Bad font weight for the labels (too thin)
- You can't delete an added antecedent
- Yes/ No button => font weight too bold

Treatment:
- ⚠️ When the name of the medicine is too long for the card

Displaying the info of added antecedents
- Yes/ No button => font weight too bold

List of antecedents:
- Bad font weight for the name of the antecedent
- The border is too wide

**Review by lucas-louis — CHANGES_REQUESTED** [review]
⚠️ If you put a bad medicine name, there is no error messages

**Review by lucas-louis — DISMISSED** [review]
LGTM

File: patient_mobile/lib/models/onboarding.dart [hunk 1]
```diff
@@ -0,0 +1,2150 @@
...
+  @override
+  State<Onboarding> createState() => _OnboardingState();
+}
+
+class _OnboardingState extends State<Onboarding> {
+  late final List<Widget> pages;
+
+  int _selectedIndex = 0;
+  final List<Map<String, dynamic>> docteurs = [
+    {'name': 'Dr. Edgar', 'address': '1 rue de la paix, 75000 Paris'},
```
Comment (SpiritGY) [inline patient_mobile/lib/models/onboarding.dart:40]: enleve les variables de test

File: patient_mobile/lib/models/onboarding.dart [hunk 2]
```diff
@@ -0,0 +1,2150 @@
...
+                if (name != "" &&
+                    lastname != "" &&
+                    birthdate != "" &&
+                    height != "" &&
+                    weight != "") {
+                  isHealths = isHealth.value;
+                  widget.updateSelectedIndex(1);
+                } else {
+                  ScaffoldMessenger.of(context).showSnackBar(ErrorLoginSnackBar(
+                      message: "Completer tout les champs", context: context));
```
Comment (SpiritGY) [inline patient_mobile/lib/models/onboarding.dart:433]: les fautes de labels, "Compléter tous les champs"

File: patient_mobile/lib/models/onboarding.dart [hunk 3]
```diff
@@ -0,0 +1,2150 @@
...
+
+  List<Map<String, dynamic>> traitments = [];
+
+  void addNewTraitement(
+      String name, Map<String, dynamic> medicines, bool stillRelevant) async {
+    setState(() {
+      traitments.add({
+        "Name": name,
+        "treatments": medicines["treatments"],
+        "Still_relevant": stillRelevant,
```
Comment (SpiritGY) [inline patient_mobile/lib/models/onboarding.dart:710]: pas de maj a "still_relevant"

File: patient_mobile/lib/services/auth.dart [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+  final response = await http.post(
+    Uri.parse(url),
+    headers: {'Content-Type': 'application/json'},
+    body: jsonEncode({
+      'email': email,
+      'password': password,
+    }),
+  );
+  Logger().i(response.body);
```
Comment (SpiritGY) [inline patient_mobile/lib/services/auth.dart:21]: logger

File: patient_mobile/lib/services/auth.dart [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+  if (response.statusCode == 200) {
+    final body = response.body;
+    final token = jsonDecode(body)['token'];
+    prefs.setString('token', token);
+    return true;
+  } else {
+    Logger().i(response.body);
```
Comment (SpiritGY) [inline patient_mobile/lib/services/auth.dart:28]: logger

File: patient_mobile/lib/services/auth.dart [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+
+Future<List<Map<String, dynamic>>> getMedecines() async {
+  await dotenv.load();
+  final url = '${dotenv.env['URL']}medicine';
+  final response = await http.get(
+    Uri.parse(url),
+  );
+  if (response.statusCode == 201) {
+    final body = response.body;
+    Logger().i(jsonDecode(body));
```
Comment (SpiritGY) [inline patient_mobile/lib/services/auth.dart:58]: logger

File: patient_mobile/lib/services/auth.dart [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+    return true;
+  } else {
+    Logger().i(response.body);
+    return false;
+  }
+}
+
+Future<Map<String, dynamic>> getTraitement() async {
+  SharedPreferences prefs = await SharedPreferences.getInstance();
+  await dotenv.load();
```
Comment (SpiritGY) [inline patient_mobile/lib/services/auth.dart:35]: faire dans la main

File: patient_mobile/lib/services/auth.dart [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+    return [];
+  }
+}
+
+Future<bool> postTraitement(Map<String, dynamic> traitement) async {
+  SharedPreferences prefs = await SharedPreferences.getInstance();
+  await dotenv.load();
+  final token = prefs.getString("token");
+  final url = '${dotenv.env['URL']}dashboard/treatment';
+  Logger().i(traitement);
```
Comment (SpiritGY) [inline patient_mobile/lib/services/auth.dart:70]: logger

File: patient_mobile/lib/services/auth.dart [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+  if (response.statusCode == 201) {
+    final body = response.body;
+    Logger().i(jsonDecode(body));
+    return List<Map<String, dynamic>>.from(jsonDecode(body)["medicament"]);
+  } else {
+    return [];
+  }
+}
+
+Future<bool> postTraitement(Map<String, dynamic> traitement) async {
```
Comment (SpiritGY) [inline patient_mobile/lib/services/auth.dart:65]: ???

File: patient_mobile/lib/services/auth.dart [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+  Logger().i(response.body);
+  Logger().i(response.statusCode);
+  if (response.statusCode == 201) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+Future<bool> postMedicalInfo(Map<String, dynamic> traitement) async {
```
Comment (SpiritGY) [inline patient_mobile/lib/services/auth.dart:85]: mauvais fichier

File: patient_mobile/lib/services/auth.dart [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+  );
+  Logger().i(response.body);
+  if (response.statusCode == 201) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+Future<List<dynamic>> getAllDoctor() async {
```
Comment (SpiritGY) [inline patient_mobile/lib/services/auth.dart:104]: mauvais fichier

File: patient_mobile/windows/flutter/CMakeLists.txt
Comment (SpiritGY) [inline patient_mobile/windows/flutter/CMakeLists.txt:1]: a garder ?

**Review by SpiritGY — CHANGES_REQUESTED** [review]

**Review by lucas-louis — DISMISSED** [review]
LGTM

File: patient_mobile/lib/screens/register.dart [hunk 12]
```diff
@@ -18,28 +29,92 @@ class Register extends StatelessWidget {
...
+            ]),
             const SizedBox(height: 20),
             PlainButton(
-                text: "Inscrivez-vous",
-                onPressed: () {
-                  Navigator.pushNamed(context, '/');
-                }),
+              text: "Inscrivez-vous",
+              onPressed: () async {
+                Logger().i("Email: $email, Password: $password");
```
Comment (SpiritGY) [inline patient_mobile/lib/screens/register.dart:72]: logger

**Review by SpiritGY — APPROVED** [review]

**Review by SpiritGY — APPROVED** [review]

**Review by lucas-louis — APPROVED** [review]

**Review by SpiritGY — APPROVED** [review]

**Review by lucas-louis — APPROVED** [review]

**Pull request merged**